src/process_request.cpp

```diff
@@ -516,7 +516,7 @@ void process_request(request &req, rate_limiter &limiter,
     }
 
     // check whether the client is being rate limited
-    if (!limiter.check(client_key)) {
+    if (!limiter.check(client_key, user_roles.count(osm_user_role_t::moderator) > 0)) {
```

Comment:
Another hint on refactoring: we have the same code in line 583. Maybe introduce a `const bool is_moderator = ... `and use that in both places.

Reply:
Fixed.